packages/flutter/lib/src/rendering/semantics_event.dart

```diff
@@ -39,9 +42,57 @@ class ScrollCompletedSemanticsEvent extends SemanticsEvent {
   /// This event should be sent after a scroll action is completed. It is
   /// interpreted by assistive technologies to provide additional feedback about
   /// the just completed scroll action to the user.
-  // TODO(goderbauer): add more metadata to this event (e.g. how far are we scrolled?).
-  ScrollCompletedSemanticsEvent() : super('scroll');
+  ///
+  /// The parameters [axis], [pixels], [minScrollExtent], and [maxScrollExtent] are
+  /// required and may not be `null`.
+  ScrollCompletedSemanticsEvent({
+    @required this.axis,
+    @required this.pixels,
+    @required this.maxScrollExtent,
+    @required this.minScrollExtent
+  }) : assert(axis != null),
+       assert(pixels != null),
+       assert(maxScrollExtent != null),
+       assert(minScrollExtent != null),
+       super('scroll');
+
+  /// The axis in which the scroll view was scrolled.
+  ///
+  /// See also [ScrollPosition.axis].
+  final Axis axis;
+
+  /// The current scroll position, in logical pixels.
+  ///
+  /// See also [ScrollPosition.pixels].
+  final double pixels;
+
+  /// The minimum in-range value for [pixels].
+  ///
+  /// See also [ScrollPosition.minScrollExtent].
+  final double minScrollExtent;
+
+  /// The maximum in-range value for [pixels].
+  ///
+  /// See also [ScrollPosition.maxScrollExtent].
+  final double maxScrollExtent;
 
   @override
-  Map<String, dynamic> toMap() => <String, dynamic>{};
+  Map<String, dynamic> toMap() {
+    final Map<String, dynamic> map = <String, dynamic>{
+      'pixels': pixels.clamp(minScrollExtent, maxScrollExtent),
```

Comment:
Should we instead `assert` that the value is within bounds?

Reply:
That assert will fire fairly often because `ScrollPosition` (where this value is usually taken from) doesn't guarantee that pixels is within bounds. Changing this to an assert would just mean that (almost) every caller would have to do the clamp before passing the value to this class. I think from an usability stand point its just simpler to do the clamp here.